File: v7_interface_segregation/java/Game2.java
package v7_interface_segregation.java;

public class Game2 {

    public static void main(String[] args) {
        Monster m = new Monster();
        m.basicAttack();
        MonsterBoss b = new MonsterBoss();
        b.magicAttack();
    }
    
}

interface BasicBadCharacterSkill {
    public void basicAttack();
}

interface AdvancedBadCharacterSkill {
    public void magicAttack();
    public void recover();
}

class Monster implements BasicBadCharacterSkill {
    public void basicAttack() {
        System.out.println("Monster basic attack");
    }
}

class MonsterBoss implements BasicBadCharacterSkill, AdvancedBadCharacterSkill {
    public void basicAttack() {
        System.out.println("Boss basic attack");
    }
    public void magicAttack() {
        System.out.println("Boss magic attack");
    }
    public void recover() {
        System.out.println("Boss recover");
    };
}

File: v7_interface_segregation/python/game2.py

class BadCharacterSkill:
    def basicAttack(self): pass

class AdvancedBadCharacterSkill:
    def magicAttack(self): pass
    def recover(self): pass

class Monster(BadCharacterSkill):
    def basicAttack(self): 
        print('monster basic attack')

class MonsterBoss(BadCharacterSkill, AdvancedBadCharacterSkill):
    def basicAttack(self): 
        print('monsterboss basic attack')

    def magicAttack(self): 
        print('monsterboss magic attack')

    def recover(self): 
        print('monsterboss recover')

m = Monster()
m.basicAttack()

mb = MonsterBoss()
mb.magicAttack()
